### ISSUE TITLE
Appscan tool reports some high severity issues against capi-release-1.77.0

### ISSUE THREAD

**Issue by Wayne-H-Ha** [cloudfoundry/capi-release#130]
Is it OK if the appscan tool reports the following high severity issues against capi-release-1.77.0:
```
Debug Flag set True in Production          cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\samples\bookinfo\src\productpage\productpage.py at line 350
Hardcoded Credentials in Python Code       cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\bin\ghdata.py at line 36
Hardcoded Credentials in Python Code       cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\release\pipeline\configs\environment_config.py at line 19
Insecure HTTP Communication                cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\golang.org\x\net\html\testdata\go1.html at line 2224
Insecure Use of Document.Write             cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\js\viz\viz-lite.js at line 1012
Insecure Use of Document.Write             cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\js\viz\viz-lite.js at line 21
Insecure Use of Document.Write             cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\js\viz\viz-lite.js at line 942
Insecure Use of InnerHTML or OuterHTML     cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\pkg\ctrlz\assets\static\js\prism.js at line 249
Insecure Use of InnerHTML or OuterHTML     cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\pkg\ctrlz\assets\static\js\prism.js at line 267
Insecure Use of InnerHTML or OuterHTML     cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\pkg\ctrlz\assets\static\js\prism.min.js at line 1
Insecure Use of InnerHTML or OuterHTML     cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\pkg\ctrlz\topics\assets\templates\metrics.html at line 130
Insecure random number                     cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\js\cola\cola.min.js at line 2
Insecure random number                     cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\js\d3\d3.v4.min.js at line 2
Insecure random number                     cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\js\viz\viz-lite.js at line 987
Insecure use of backticks                  cloud_controller_ng\db\migrations\20130603223751_convert_latin1_to_utf8_for_mysql.rb at line 5
Insecure use of backticks                  cloud_controller_ng\db\migrations\20150514190458_fix_mysql_collations.rb at line 19
Insecure use of backticks                  cloud_controller_ng\db\migrations\20150514190458_fix_mysql_collations.rb at line 30
Insecure use of backticks                  cloud_controller_ng\lib\cloud_controller\db_connection\options_factory.rb at line 12
Insecure use of backticks                  cloud_controller_ng\lib\cloud_controller\encryptor.rb at line 120
JQuery HTML Function Use                   cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\force\forcegraph.js at line 246
JQuery Insecure Manipulation of Child Node cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\force\forcegraph.js at line 66
JQuery Insecure Manipulation of Child Node cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\force\forcegraph.js at line 72
Join and relpath path traversal            cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\bin\boilerplate\boilerplate.py at line 158
Join and relpath path traversal            cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\bin\boilerplate\boilerplate.py at line 176
PostMessage To Any Target Origin           cloud_controller_ng\vendor\cache\copilot-6a31452d0423\vendor\istio.io\istio\addons\servicegraph\js\viz\viz-lite.js at line 967
```

**Comment by cwlbraa** [comment]
the only interesting ones (ie the only ones that are not in historical migrations or transitively-vendored copilot->istio code) are these 2:

```Insecure use of backticks                 
cloud_controller_ng\lib\cloud_controller\db_connection\options_factory.rb at line 12
Insecure use of backticks                  cloud_controller_ng\lib\cloud_controller\encryptor.rb at line 120```

**Comment by tcdowney** [comment]
@cwlbraa I updated those two lines to use single-quotes instead of backticks because there isn't a good reason to use the backticks. The input was coming either from salt column names that we control or database configuration from operator-supplied BOSH config and only being used in error messages, but rather not worry about scanners picking it up.

@Wayne-H-Ha 
The old migrations are using them in a SQL context and are also no operating on user-submitted input so those should be fine as well.

The Istio stuff surprised me. That's not actually active code, but it looks like since https://github.com/cloudfoundry/cloud_controller_ng/commit/d8d9963f307f485d3f1b075ab46a750ea3de9101 `bundler` is pulling down and vendoring the whole https://github.com/cloudfoundry/copilot repo instead of just the small portion of it that is their Ruby SDK. 😒

Starting a conversation with the networking team to see if they're still getting value from that, or if we can move back to using a released version of the `cf-copilot` gem. Wonder if this explains some of the huge amounts of file-traversal folks have reported on start up (like in #125).

[Slack convo](https://cloudfoundry.slack.com/archives/CFX13JK7B/p1552953939151100)

**Comment by tcdowney** [comment]
The Networking team cut a new release of the `cf-copilot` gem and we're now using it as of https://github.com/cloudfoundry/cloud_controller_ng/commit/e79821e7d5502aa83d5605d7ebf1daaf145a08f3. This should cut down on the number and size of files we're vendoring.

**tl;dr**
None of the things that the scan picked up were actually security issues, but it did point out that the way we were installing the `cf-copilot` gem was unnecessarily bloating the size of capi-release.

Gonna go ahead and close this issue, thanks @Wayne-H-Ha!
